src/provider-session-manager.ts

```
@@ -54,12 +54,25 @@ export class ProviderSessionManager {
+      if (err instanceof Error && err.message === 'Authorization timed out') {
+        throw new Error('Could not complete GitHub authentication flow');
+      }
+    }
+
+    if (!newAuthSession) {
+      extensionApi.window.showNotification({title: 'Could not complete GitHub authentication flow', type: 'error', highlight: true});
+      throw new Error('Could not complete GitHub authentication flow');
     } else {
-      throw new Error('Could not complete authentication flows');
+      extensionApi.window.showNotification({title: 'Successfully authenticated with GitHub', type: 'info', highlight: true});
     }
```

Biome (2.1.2)

[error] This else clause can be omitted because previous branches break early.

Safe fix: Omit the else clause.


(lint/style/noUselessElse)